[FLINK-14068][streaming] Removes deprecated org.apache.flink.streaming.api.windowing.time.Time

XComp opened the pull request:
## What is the purpose of the change

Removes deprecated `org.apache.flink.streaming.api.windowing.time.Time`

## Brief change log

The changes where done using the IDE refactoring and find&replace operations.

## Verifying this change

* CI should pass

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

XComp commented:
Converting this one to draft. Things to do:
[ ] Going through the code diff and identifying wrong/missing code changes
[ ] Going through the docs diff and identifying wrong/missing changes

XComp commented on flink-streaming-scala/src/main/scala/org/apache/flink/streaming/api/scala/AllWindowedStream.scala:
```diff
@@ -70,7 +71,7 @@ class AllWindowedStream[T, W <: Window](javaStream: JavaAllWStream[T, W]) {
    */
   @PublicEvolving
-  def allowedLateness(lateness: Time): AllWindowedStream[T, W] = {
+  def allowedLateness(lateness: Duration): AllWindowedStream[T, W] = {
```
There are scala changes like that that didn't include proper deprecation because we wanted to remove scala entirely, anyway, in 2.0.

What's the state with that?

zentol commented on docs/content.zh/docs/ops/state/task_failure_recovery.md:
```diff
@@ -172,10 +172,10 @@ restart-strategy.exponential-delay.attempts-before-reset-backoff: 10
 env.setRestartStrategy(RestartStrategies.exponentialDelayRestart(
-  Time.milliseconds(1),
-  Time.milliseconds(1000),
+  Duration.ofMillis(1),
```
I think this is wrong; this here uses the common Time class.

zentol replied:
This wasn't addressed yet?

zentol commented on docs/content/docs/ops/state/task_failure_recovery.md:
```diff
@@ -188,10 +188,10 @@ StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironm
 env.setRestartStrategy(RestartStrategies.exponentialDelayRestart(
-  Time.of(1, TimeUnit.MILLISECONDS), // initial delay between restarts
-  Time.of(1000, TimeUnit.MILLISECONDS), // maximum delay between restarts
+  Duration.ofMillis(1), // initial delay between restarts
```
I think this is wrong; this here uses the common Time class.

zentol commented on flink-runtime/src/test/java/org/apache/flink/streaming/runtime/operators/windowing/TimeWindowTranslationTest.java:
```diff
@@ -79,22 +78,17 @@ void testAlignedWindowDeprecation() {
         DataStream<Tuple2<String, Integer>> window2 =
                 source.keyBy(0)
-                        .timeWindow(Time.of(1000, TimeUnit.MILLISECONDS))
+                        .windowAll(TumblingProcessingTimeWindows.of(Duration.ofMillis(1000)))
```
it's weird this uses an AllWindow when we have a keyed stream.

zentol commented on flink-streaming-scala/src/main/scala/org/apache/flink/streaming/api/scala/CoGroupedStreams.scala:
```diff
@@ -51,7 +52,7 @@ import scala.collection.JavaConverters._
  *     .where(new MyFirstKeySelector())
  *     .equalTo(new MyFirstKeySelector())
- *     .window(TumblingEventTimeWindows.of(Time.of(5, TimeUnit.SECONDS)))
+ *     .window(TumblingEventTimeWindows.of(Time.ofSeconds(5)))
```
Duration.ofSeconds

zentol commented on flink-streaming-scala/src/test/scala/org/apache/flink/streaming/api/scala/TimeWindowTranslationTest.scala:
```diff
@@ -68,13 +66,13 @@ class TimeWindowTranslationTest extends AbstractTestBaseJUnit4 {
-      .apply(new WindowFunction[(String, Int), (String, Int), Tuple, TimeWindow]() {
-        def apply(
-            key: Tuple,
+      .windowAll(TumblingProcessingTimeWindows
```
same concern as elsewhere about windowAll on a keyed stream

zentol reviewed (commented):
Nice, looks mostly fine, but had a few comments.

XComp commented:
Most of the Scala code is gone in `master` now which unblocked this PR (I had an issue with the Scala compiler fixing one of the windowAll comments and I didn't want to bother to fix it considering that we were planning to remove the Scala code). I rebased the branch and addressed all comments.

XComp commented:
Failure related to FLINK-36356.

XComp commented:
@flinkbot run azure

zentol commented on README.md:
```diff
@@ -41,7 +41,7 @@ val text = env.socketTextStream(host, port, '\n')
 val windowCounts = text.flatMap { w => w.split("\\s") }
   .map { w => WordWithCount(w, 1) }
   .keyBy("word")
-  .window(TumblingProcessingTimeWindow.of(Time.seconds(5)))
```
remove scala example?

XComp replied:
I will leave that to [FLINK-36352](https://issues.apache.org/jira/browse/FLINK-36352)

zentol commented on docs/content.zh/docs/dev/datastream/operators/overview.md:
````diff
@@ -198,19 +198,19 @@ data_stream.key_by(lambda x: x[1]).reduce(lambda a, b: (a[0] + b[0], b[1]))
 ```scala
 dataStream
   .keyBy(_._1)
-  .window(TumblingEventTimeWindows.of(Time.seconds(5))) 
````
why are the docs still full of scale examples if the apis were removed?

XComp replied:
FLINK-36352

zentol reviewed (approved):
Bit irritated that we got scale examples in the docs, but it's not a problem for this PR I suppose.

XComp commented:
> Bit irritated that we still got scala examples in the docs, but it's not a problem for this PR I suppose.

Looks like there is a separate issue FLINK-36352 for that.

XComp commented:
@flinkbot run azure

XComp commented:
@flinkbot run azure

XComp commented:
Something's odd with the cibot. But GHA had a successful CI run on this PR/commit: https://github.com/XComp/flink/actions/runs/11091729815

I will merge the PR. Thanks for the approval
